packages/fast-usdc/README.md

```diff
@@ -59,43 +59,39 @@ sequenceDiagram
 
 # Status Manager
 
-### Pending Advance State Diagram
+### State Diagram
 
-*Transactions are qualified by the OCW and EventFeed before arriving to the Advancer.*
+*Transactions are qualified by the OCW and TransactionFeed before being
+delivered to the Advancer.*
 
-```mermaid
-stateDiagram-v2
-  [*] --> Observed: observe()
-  [*] --> Advancing: advancing()
-
-  Advancing --> Advanced: advanceOutcome(...true)
-  Advancing --> AdvanceFailed: advanceOutcome(...false)
-
-  Observed --> [*]: dequeueStatus()
-  Advanced --> [*]: dequeueStatus()
-  AdvanceFailed --> [*]: dequeueStatus()
-
-  note right of [*]
-    After dequeueStatus():
-    Transaction is removed 
-    from pendingTxs store.
-    Settler will .disburse()
-    or .forward()
-  end note
-```
+The transactionFeed receives attestations from Oracles, records their
+evidence, and when enough oracles agree, (if no risks are identified)
+it publishes the results for the advancer to act on.
+
+The Advancer subscribes (via `handleTransactionEvent`) to events published by
+the transactionFeed. When notified of an appropriate opportunity, it is
+responsible for advancing funds to fastUSDC payees.
 
-### Complete state diagram (starting from Transaction Feed into Advancer)
+The Settler is responsible for monitoring (via `receiveUpcall`) deposits to the
+settlementAccount. It either `disburse`s funds to the Pool (if funds were
+`advance`d to the payee), or `forwards` funds to the payee (if pool funds
+were not `advance`d).
 
 ```mermaid
 stateDiagram-v2
+  state Forwarding <<choice>>
+  state AdvancingChoice <<choice>>
+
   Observed --> AdvanceSkipped : Risks identified
   Observed --> Advancing : No risks, can advance
   Observed --> Forwarding : No risks, Mint deposited before advance
```

Comment:
> Observed --> Forwarding : No risks, Mint deposited before advance

We should expand this more because it's no longer accurate. An invariant is that the contract never `Forward`s without seeing evidence. When `Settler.receiveUpcall` fires and there are no matches in `StatusManager`'s `pendingTxs`, it goes into the `mintedEarly` store. Each time `Advancer.handleTransactionEvent` is invoked, it will call `Settler.notifier.checkMintedEarly()` and early return if true. `checkMintedEarly` will clear the item from `mintedEarly` and initiate a `.forward()`.

Reply:
I agree that all that happens, and it complicates our process, but if it complicates the state transitions, it's not obvious to me how it should be represented. AFAICT, the state transitions are unaffected by all of this. 

Should we represent those possibilities more explicitly in the states? 

Perhaps I could add a note on the **choice** node listing the conditions that the code pays attention to to decide whether and when to attempt to forward.